### ISSUE TITLE
Ui router is not working for the first click and works for the proceeding clicks

### ISSUE THREAD

**Issue by Jeevanandanj** [angular-ui/ui-router#2635]
I am using angular 1.3.7 and ui router v0.2.18.

I've migrated from ng route to ui route.
Now I removed ng-href and window.location.hash and used ui-sref and $state.transitionTo and $state.go and $state.reload based on need.

I am using ocLazyload to load js files.

Strange is when i login to my application at first every links are working.

But I have implemented 3 new pages (3 states), after navigate to this pages,

**then each and every state is not triggering at first click but after first click each state, then i can navigate to visited state at first click.**

**Even for the first click, $stateChangeStart has been triggered but url is not changed.**

No error has been logged.
I don't know the root cause for this issue. 
What will be the solution for this?

My routing

```
.state('home.T_BoardWidgets', {
            url: '/T_BoardWidgets',
            title: 'T_BoardWidgets',
            authenticate: true,
            controller: 'BoardWidgetController',
            caseInsensitiveMatch: true,
            templateUrl: baseUrl + '/App/Board/_boardWidgets.html?v=' + version,
            resolve: {
                loadModule: ['$q', '$ocLazyLoad', function ($q, $ocLazyLoad) {
                    var deferred = $q.defer();
                    $ocLazyLoad.load(['App/Board/boardService.js?v=' + version,
                                        'scripts/angular-img-cropper/angular-img-cropper.js',
                                          'scripts/angular-file-upload.js',
                                          'App/Board/BoardWidgetCtrl.js?v=' + version], { serie: true }).then(function () {
                                              $ocLazyLoad.inject('angularFileUpload');
                                              $ocLazyLoad.inject('ngImgCrop');
                                              deferred.resolve();
                                          });

                    return deferred.promise;
                }],
            }
        })
```

**Comment by Jeevanandanj** [comment]
Found the issue root cause.

The ui router is amazing. But the small issue has to be solved.

Please consider the following issue.

**Root cause : **
Injected the unknown angular module  $ocLazyLoad.inject('ngImgCrop');.
So this has provided the error.

**Issue**
After the issue, the state has been loaded for the second click forcefully.
But the working state (which is previously worked earlier), that has also been affected because of this state.

The issue is, even worked states are worked only after the second click.

**Need a fix:**
After face the exception, The exception is not catched in $stateChangeError.
This has been not cached for past two days.

I don't know, how the issue is cached now.

Instead of deferred.resolve,

I've used like  `return  $ocLazyLoad.load`, after this the exception has been cached. Again I used deferred.resolve, then again exception has been cached.

Note: Tomorrow, I'll update the plunkr.

**Comment by christopherthielen** [comment]
> Note: Tomorrow, I'll update the plunkr.

Cool, we'll wait for your plunker that demonstrates the error

**Comment by eddiemonge** [comment]
closing due to inactivity
